api/rpc/client/client.go

```diff
@@ -0,0 +1,42 @@
+package client
+
+import (
+	"context"
+
+	"github.com/filecoin-project/go-jsonrpc"
+
+	"github.com/celestiaorg/celestia-node/nodebuilder/daser"
+	"github.com/celestiaorg/celestia-node/nodebuilder/fraud"
+	"github.com/celestiaorg/celestia-node/nodebuilder/header"
+	"github.com/celestiaorg/celestia-node/nodebuilder/share"
+	"github.com/celestiaorg/celestia-node/nodebuilder/state"
+)
+
+//go:generate go run github.com/golang/mock/mockgen -destination=../../mocks/api.go -package=mocks . API
+type API interface {
```

Comment:
this interface is generated for mocking? Can you store these in a separate file, maybe `mock.go` per package? I think that's more standard.

Reply:
So, they are stored in a separate file - the `go:generate` uses this API interface to autogenerate the mocks used in `api/mocks/api.go`

Reply:
ack, resolve.